Sources/Maintenance/Maintenance.swift

```diff
@@ -0,0 +1,124 @@
+//
+//  File.swift
+//  
+//
+//  Created by Emad Ghorbania on 14/12/2022.
+//
+
+import Foundation
+import FirebaseRemoteConfig
+import Helpers
+
+struct Maintenance {
+    var fetch: (_ minimumFetchInterval: Double?,
+                _ bannerKey: String,
+                _ fullscreenKey: String,
+                _ bannerDetailsKey: String,
+                _ fullscreenDetailsKey: String) async throws -> (StatusType, Detail)?
+    
+    public static let live = Self { minimumFetchInterval, bannerKey, fullscreenKey, bannerDetailsKey, fullscreenDetailsKey in
+        try await Maintenance.fetchFromFirebaseRemoteConfige(
+            minimumFetchInterval: minimumFetchInterval,
+            bannerKey: bannerKey,
+            fullscreenKey: fullscreenKey,
+            bannerDetailsKey: bannerDetailsKey,
+            fullscreenDetailsKey: fullscreenDetailsKey)
+    }
+    /// You should initiate this struct by injecting variables to be able to start using ``Maintenance`` and it Validates the latest version of the app!
+    /// - Warning:You should configure FirebaseApp in your project by injecting info.plist to them before using this module
+    /// - Example:
+    /// ````
+    /// do {
+    ///     let maintenance = try await Maintenance.fetchFromFirebaseRemoteConfige(bannerKey: "Test_Banner_iOS",
+    ///                                                                            fullscreenKey: "Test_Fullscreen_iOS",
+    ///                                                                            bannerDetailsKey: "Test_BannerDetail_iOS",
+    ///                                                                            fullscreenDetailsKey: "Test_FullscreenDetail_iOS")
+    ///     switch maintenance?.0 {
+    ///     case .banner:
+    ///         print(maintenance?.1)
+    ///     case .fullscreen
+    ///         print(maintenance?.1)
+    ///     case .none:
+    ///         No Maintenance
+    ///     }}
+    /// } catch {
+    ///     Send error to crashlytics
+    /// }
+    ///````
+    /// - Parameters:
+    ///   - bannerKey: The Firebase Remote Config key for banner
+    ///   - fullscreenKey: The Firebase Remote Config key for fullscreen
+    ///   - bannerDetailsKey: The Firebase Remote Config key for bannerDetails
+    ///   - fullscreenDetailsKey: The Firebase Remote Config key for fullscreenDetails
```

Comment:
From the description Im not sure what value does these keys point at.
Are the keys pointing a string values that can be shown in the app?

Reply:
will add in next commit